### DISCUSSION TITLE
Are "vercel dev" and Turborepo incompatible?

### DISCUSSION THREAD

**Discussion by gsheppard** [vercel/vercel#8656] [Help] [answered]
I'm using Turborepo to manage a multi-tenant app. Development hinges on being able to run all Next apps at the same time, which why I went with Turborepo in the first place.

For managing environment variables, I'd like to use Vercel's implementation. But it requires using `vercel dev` locally. But `vercel dev` can only be run against 1 project at a time, kind of nullifying Turborepo.

Is there something I'm missing on how best to apply per-app environment variables in development when using the two technologies?

**Comment by gsheppard** [comment, marked as answer]
I didn't realize `vercel pull` and `vercel env pull` did entirely different things. `vercel env pull` is definitely what I need to be more compatible with `dotenv`.

Thanks and sorry!

**Comment by nathanhammond** [comment]
Heya @gsheppard! You're correct that `vercel dev` and Turborepo don't play nicely together. `vercel dev` wants to spin up a single application, and `turbo run build` wants to run as much as possible at once. This is something we're aware of and working on a better story for.
